rewrite_external_links/middleware.py

```diff
@@ -21,30 +21,32 @@ class RewriteExternalLinksMiddleware(object):
         <a href="/external_link/?link=http://www.xxx.com&next=...">
     """
 
-    #                            <a ... href="                www.xxx.com                                " ... />
-    extlinks = re.compile(r'''(?P<before><a[^>]*href=['"]?)(?P<link>https?://''' + safe_urls + '''[^'">]*)(?P<after>[^>]*)''')
+    # <a ... href="    www.xxx.com                                " ... />
+    extlinks = re.compile(r'''
+        (?P<before><a[^>]*href=['"]?)
+        (?P<link>https?://{}[^'">]*)
+        (?P<after>[^>]*)
+    '''.format(safe_urls), re.VERBOSE)
```

Comment:
You can now move the comments inline with the regex:

```python
extlinks = re.compile(r'''
    (?P<before><a[^>]*href=['"]?)  # <a ... href="
    (?P<link>https?://{}[^'">]*)  # www.xxx.com
    (?P<after>[^>]*)  # " ... />
'''.format(safe_urls), re.VERBOSE)
```

I'm not sure the comments are in sync though...